.github/workflows/mobile-android_build.yml

```diff
@@ -58,7 +58,7 @@ jobs:
       contents: read
       packages: read
     name: java_helloworld
-    runs-on: macos-12
+    runs-on: mobile
```

Comment:
I know this is the current name of this runner, but I wonder if we should cook up a name that at least has linux in the name? But that's out of scope for this PR, obviously.

Reply:
+1. I also think we should call this something like `ubuntu-22.04-large` or something like that.

Reply:
not sure its mobile specific

envoy-x64-$SIZE would be our current convention as used in azp

Reply:
the envoy prefix is because the pools are shared (subject to restrictions etc) across the org so if any other projects require separate runners etc (eg nighthawk) then we know which is which

i would suggest fixing this before landing - its a pita changing the name once PRs are being raised against it

Reply:
I see we settled on `envoy-x64-small` for this runner. I'm surprised that we don't have the OS in the name. This makes it harder for us to tell at a glance which CI jobs are using which platforms (and the info on the runners is hidden in github UI that we don't have access to). Can we revisit this naming scheme? Alternatively, maybe we could publish info on our runners in some doc somewhere?

Reply:
>  I'm surprised that we don't have the OS in the name.

no point really - all of the ms hosted linux are ubuntu - and similar with azp and our self-hosted

> This makes it harder for us to tell ...

assume linux/ubuntu and all will be clear

> Alternatively, maybe we could publish info on our runners in some doc somewhere?

happy to review - but its something that changes over time with necessity/dev so it would add maintenance - the ci yaml source is kinda the authority here

perhaps we can add a high level - this is _how_ they are named doc

Reply:
> > I'm surprised that we don't have the OS in the name.
> 
> no point really - all of the ms hosted linux are ubuntu - and similar with azp and our self-hosted

So some runners are macOS, some are linux and (pending the Windows work) some will be windows. This seems like an important detail that would be easy to encode in the runner name without causing any problems. Looking at the "runs-on" fields in our .yml files today we have:
```
     14     runs-on: ${{ needs.env.outputs.agent_ubuntu }}
     18     runs-on: macos-12
      5     runs-on: ubuntu-20.04
     21     runs-on: ubuntu-22.04
```

All of them seem to have the OS in the name, and this seems really useful.
 
> > This makes it harder for us to tell ...
> 
> assume linux/ubuntu and all will be clear

Why assume when we could be explicit? I think it would be better to have a naming scheme that includes the OS.

> > Alternatively, maybe we could publish info on our runners in some doc somewhere?
> 
> happy to review - but its something that changes over time with necessity/dev so it would add maintenance 

If we have a naming scheme which includes the relevant info then there is no need for any docs, of course. That seems like the least burden.

> - the ci yaml source is kinda the authority here

Oh, is there a yaml file I could look at which would tell me that `envoy-x64-small` is linux? If so, that's great.

> perhaps we can add a high level - this is _how_ they are named doc

That would be great too.

Reply:
> So some runners are macOS, some are linux and (pending the Windows work) some will be windows. This seems like an important detail that would be easy to encode in the runner name without causing any problems. Looking at the "runs-on" fields in our .yml files today we have:

thas kinda the point in a way - macos version **determines** the arch, windows is only x64 

linux is the only one with variants, and they are **all** ubuntu

this is how we came to the naming scheme that we have used so far (cc @mum4k ) and tbh i think it makes sense - no need to add redundant info

> Oh, is there a yaml file I could look at which would tell me that envoy-x64-small is linux? If so, that's great.

no but you can do `git grep runs-on` to find out what is currently used

Reply:
> > So some runners are macOS, some are linux and (pending the Windows work) some will be windows. This seems like an important detail that would be easy to encode in the runner name without causing any problems. Looking at the "runs-on" fields in our .yml files today we have:
> 
> thas kinda the point in a way - macos version **determines** the arch, windows is only x64
> 
> linux is the only one with variants, and they are **all** ubuntu
> 
> this is how we came to the naming scheme that we have used so far (cc @mum4k ) and tbh i think it makes sense - no need to add redundant info
> 
> > Oh, is there a yaml file I could look at which would tell me that envoy-x64-small is linux? If so, that's great.
> 
> no but you can do `git grep runs-on` to find out what is currently used

Yes, that's what I did and you'll notice that all the existing runners (except the new one we're talking about here) have the OS in the name:
* macos-12
* ubuntu-20.04
* ubuntu-22.04
The only runner that does NOT have the OS name is `envoy-x64-small`. I'm suggesting that we stick `linux` or `ubunutu` (or `ubuntu-22.04`) in the name. I don't understand what the upside is to NOT having the OS in the name, especially since the GH runner names all do.

Reply:
that is the ms-hosted naming - we made up our own when managing self-hosted pools for various reasons

apart from anything _not_ doing that emphasizes these are not the standard vms

Reply:
I don't think I understand. The current runners in the yaml file have OS in the name. For at least mobile developers, having the OS in the runner name makes it easier to reason about our CI jobs. The new runner does not have OS in the name. I think we should rename it to include the OS in the name (`envoy-x64-linux-small`, `envoy-x64-ubuntu-small`, `envoy-x64-ubuntu-22.04-4cpu-16gb`. Can we just do that? Or is there some reason it would be *bad* to have the OS in the name? If there is, I don't think I understand what it is.

Reply:
> The current runners in the yaml file have OS in the name

only the ms-hosted ones

i think its generally useful to be able to discriminate them in that way

Reply:
> > The current runners in the yaml file have OS in the name
> 
> only the ms-hosted ones

Are there any runners in the current yaml file `runs-on:` which do not have OS in the name?
 
> i think its generally useful to be able to discriminate them in that way

Do I understand correctly that you think we should NOT put OS in the name of the github runners we create because the MS runners have the OS name and so if a runner does not have the name we know it's a custom GH runner? I'd certainly be happy to see some token in the name to discriminate between MS-hosted runners and GH-custom runners (perhaps "-ms" and "-gh", but I'm game for anything). But the lack of OS in the name is not compelling.

But maybe I misunderstanding your point.

Reply:
> Are there any runners in the current yaml file runs-on: which do not have OS in the name?

yes - all of the "custom" ones

> Do I understand correctly that you think we should NOT put OS in the name of the github runners we create because the MS runners have the OS name and so if a runner does not have the name we know it's a custom GH runner?

yes and other than in the case of mac or win - the platform is superfluous its **always** linux and ubuntu

Reply:
and in the case of mac or win the arch is determined

so in those cases - as you can see with current windows image - i have named the first pool `envoy-win19-medium` 

as that has **all** of the salient information with the caveat that we could/should define somewhere what small/medium/large/etc means (and i guess where thay can be used)

Reply:
... and `envoy-win22-medium` (for ref)

Reply:
> > Are there any runners in the current yaml file runs-on: which do not have OS in the name?
> 
> yes - all of the "custom" ones

What's an example of what you're talking about? I don't seem to be able to find them:

```
% git grep -h 'runs-on:'  | sort | uniq -c | sort -nr       
     21     runs-on: ubuntu-22.04
     14     runs-on: macos-12
     14     runs-on: ${{ needs.env.outputs.agent_ubuntu }}
      5     runs-on: ubuntu-20.04
      4     runs-on: envoy-x64-small
      1       runs-on: envoy-x64-small
      1     runs-on: ${{ inputs.runs-on }}
      1       runs-on:
```
 
> > Do I understand correctly that you think we should NOT put OS in the name of the github runners we create because the MS runners have the OS name and so if a runner does not have the name we know it's a custom GH runner?
> 
> yes and other than in the case of mac or win - the platform is superfluous its **always** linux and ubuntu

It's clearly useful to know if the runner is mac, win or linux. I don't think there is any reason to omit this information just for custom github linux runners.

Reply:
> `envoy-win22-medium`

This seems like a great name. Let's just add linux or ubunutu to the new runner to match that pattern (and/or remove x64. I don't have any opinion about that)

Reply:
i dont agree for the reasons explained

Reply:
@alyssawilk when you're back, can you please tie-break here.

Reply:
you win - happy to review